pineappl/src/interpolation.rs

```diff
@@ -641,6 +656,32 @@ mod tests {
         }
     }
 
+    #[test]
+    fn compare_fields_with_nan() {
+        let interp_a = Interp::new(
+            1e-3,
+            1e4,
+            50,
+            3,
+            ReweightMeth::NoReweight,
+            Map::ApplGridH0,
+            InterpMeth::Lagrange,
+        );
+        let interp_b = Interp::new(
+            1e-3,
+            1e4,
+            50,
+            3,
+            ReweightMeth::NoReweight,
+            Map::ApplGridH0,
+            InterpMeth::Lagrange,
+        );
```

Comment:
Delete `interp_b` and compare `interp` with itself, that should make the intent clearer.